app/controllers/cms_controller.rb

```diff
@@ -36,11 +22,51 @@ def clear_page_cache
     redirect_to request.fullpath.sub(%r{/refresh$}, "")
   end
 
+  def privacy
+    process_resource Cms::Pages::PrivacyNotice
+  end
+
+  def deep_test
+    process_resource Cms::Pages::DeepTest
+  end
+
   private
 
+  def process_collection(cls, title: "News & Updates", page_name: "Articles", collection_wrapper: "ncce-news-archive")
+    page =
+      if params[:page].present?
+        params[:page].to_i
+      else
+        1
+      end
+    @title = title
+    @page_name = page_name
+    @collection_wrapper_class = collection_wrapper
+    @path = cms_posts_path
+    @collection = cls.all(page, 50)
+    render :collection
+  end
+
+  def process_resource(cls, resource_params: {})
```

Comment:
What other things can I put in resource_params other than the resource_id? If its just the resource_id could we pass that around rather than the hash?

Reply:
It should only be the ID now, I think this had more a role before I refactored, I will update this to just be the ID, as it will be clearer whats its role is